src/main.rs

```diff
@@ -28,12 +30,20 @@ fn main() -> Result<(), &'static str> {
             Message::ExitStatusSuccess => {
                 success_count += 1;
             }
-            Message::ExitStatusFailure => {
-                failure_count += 1;
+            Message::ExitStatusFailure(code) => {
+                if opt.exit_early_on_error {
+                    process::exit(code.unwrap_or(1));
+                } else {
+                    failure_count += 1;
+                }
             }
             Message::FailedToRun(error) => {
-                failure_count += 1;
-                bar.println(error);
+                if opt.exit_early_on_error {
+                    process::exit(1);
+                } else {
+                    failure_count += 1;
+                    bar.println(error);
```

Comment:
I think this error should probably be printed in both cases

Reply:
Done -- apologies for the delay!